src/dotnet/commands/dotnet-add/dotnet-add-package/Program.cs

```diff
@@ -0,0 +1,176 @@
+               LocalizableStrings.CmdNoRestoreDescription,
+               CommandOptionType.NoValue);
+
+            command._sourceOption = command.Option(
+                $"-s|--source <{LocalizableStrings.CmdSource}>",
+                LocalizableStrings.CmdSourceDescription,
+                CommandOptionType.SingleValue);
+
+            command._packageDirectoryOption = command.Option(
+                $"--package-directory <{LocalizableStrings.CmdPackageDirectory}>",
+                LocalizableStrings.CmdPackageDirectoryDescription,
+                CommandOptionType.SingleValue);
+
+            return command;
+        }
+
+        public override int Run(string fileOrDirectory)
+        {
+            if (RemainingArguments.Count != 1)
+            {
+                throw new GracefulException(LocalizableStrings.SpecifyExactlyOnePackageReference);
+            }
+
+            var projectFilePath = string.Empty;
+
+            if (!File.Exists(fileOrDirectory))
+            {
+                projectFilePath = MsbuildProject.GetProjectFileFromDirectory(fileOrDirectory).FullName;
+            }
+            else
+            {
+                projectFilePath = fileOrDirectory;
+            }
+
+            var tempDgFilePath = string.Empty;
+
+            if (!_noRestoreOption.HasValue())
+            {
+                // Create a Dependency Graph file for the project
+                tempDgFilePath = Path.GetTempFileName();
+                GetProjectDependencyGraph(projectFilePath, tempDgFilePath);
+            }
+
+            var result = NuGetCommand.Run(TransformArgs(RemainingArguments.First(), tempDgFilePath, projectFilePath));
+            DisposeTemporaryFile(tempDgFilePath);
+
+            return result;
+        }
+
+        private void GetProjectDependencyGraph(string projectFilePath, string dgFilePath)
+        {
+            var args = new List<string>();
+
+            // Pass the project file path
+            args.Add(projectFilePath);
+
+            // Pass the task as generate restore Dependency Graph file
+            args.Add("/t:GenerateRestoreGraphFile");
+
+            // Pass Dependency Graph file output path
+            args.Add($"/p:RestoreGraphOutputPath=\"{dgFilePath}\"");
+
+            var result = new MSBuildForwardingApp(args).Execute();
+
+            if (result != 0)
+            {
+                throw new GracefulException(string.Format(LocalizableStrings.CmdDGFileException, projectFilePath));
+            }
+        }
+
+        private void DisposeTemporaryFile(string filePath)
+        {
+            if (File.Exists(filePath))
+            {
+                File.Delete(filePath);
+            }
+        }
+
+        private string[] TransformArgs(string packageId, string tempDgFilePath, string projectFilePath)
+        {
+            var args = new List<string>(){
+                "package",
+                "add",
+                "--package",
+                packageId,
```

Comment:
We do - https://github.com/mishra14/cli/blob/ac988a4ac92cecb3eb21a58e88cf645f0f2d7279/src/dotnet/commands/dotnet-add/dotnet-add-package/Program.cs#L143